addons/mail/static/src/core/common/message_link_preview_list.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-name="mail.MessageLinkPreviewList">
+        <div class="o-mail-LinkPreviewList d-flex flex-column mt-2" t-att-class="{ 'me-2 pe-4': env.inChatWindow and !env.alignedRight, 'ms-2 ps-4': env.inChatWindow and env.alignedRight }">
+            <div class="d-flex flex-grow-1 flex-wrap">
+                <t t-if="props.messageLinkPreviews.length > 1">
+                    <t t-foreach="props.messageLinkPreviews" t-as="messageLinkPreview" t-key="messageLinkPreview.id">
+                        <LinkPreview linkPreview="messageLinkPreview.link_preview_id" delete.bind="messageLinkPreview.hide" deleteAll.bind="messageLinkPreview.message_id.hideAllLinkPreviews"/>
+                    </t>
+                </t>
+                <t t-else="">
+                   <LinkPreview linkPreview="props.messageLinkPreviews[0].link_preview_id" delete.bind="props.messageLinkPreviews[0].hide"/>
+                </t>
+            </div>
```

Comment:
Note that `LinkPreview` is mostly based on `mail.link.preview` except for customised `delete` and `deleteAll` props. But we could also turn the component into flavoured `mail.message.link.preview` and have no props (and this contains the very specific context).

Not a big deal, but I think reducing amount of props help reducing complexity, as the features are all in the same nest (the component inner-logic).

Reply:
Reusable components are a good thing and I think this one is a good one.